modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/filename/AbstractDataRegionRelativeStoragePathTest.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.persistence.filename;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.ObjIntConsumer;
+import java.util.stream.IntStream;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.cache.affinity.rendezvous.RendezvousAffinityFunction;
+import org.apache.ignite.cluster.ClusterState;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.DataRegionConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ * Test cases when {@link DataRegionConfiguration#setStoragePath(String)} used to set custom data region storage path.
+ */
+@RunWith(Parameterized.class)
+public abstract class AbstractDataRegionRelativeStoragePathTest extends GridCommonAbstractTest {
+    /** Custom storage path for default data region. */
+    static final String DEFAULT_DR_STORAGE_PATH = "dflt_dr";
+
+    /** Custom storage path for custom data region. */
+    static final String CUSTOM_STORAGE_PATH = "custom_dr";
+
+    /** Data region name with custom storage. */
+    static final String DR_WITH_STORAGE = "custom-storage";
+
+    /** Data region with default storage. */
+    static final String DR_WITH_DFLT_STORAGE = "default-storage";
+
+    /** */
+    static final String SNP_PATH = "ex_snapshots";
+
+    /** */
+    @Parameterized.Parameter()
+    public boolean useAbsStoragePath;
+
+    /** */
+    @Parameterized.Parameters(name = "useAbsStoragePath={0}")
+    public static List<Object[]> params() {
```

Comment:
This does not need to be wrapped in a collection for a single parameter - it is enough to return `return new Object[]{true, false}`